src/main/java/seedu/address/logic/commands/InvokeCommand.java

```diff
@@ -0,0 +1,40 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.logic.commands.exceptions.InvokeCommandException;
+import seedu.address.logic.parser.CommandRecorder;
+import seedu.address.model.Model;
+
+/**
+ * Invoke the most recent command that was typed in by user.
+ * Invoke will not count "invoke" as a result, it will return the most recent command that is not "invoke".
+ */
+public class InvokeCommand extends Command {
+
+    public static final String COMMAND_WORD = "invoke";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Invoke the most recently used command "
+            + "Parameters: none\n"
+            + "Example: " + COMMAND_WORD;
+
+    public InvokeCommand() {}
+
+    @Override
+    public CommandResult execute(Model model) throws InvokeCommandException {
+        requireNonNull(model);
+        throw new InvokeCommandException(CommandRecorder.getLastCommand());
+    }
```

Comment:
This method always throws an exception. Would this be a better implementation?
```java
@Override
public CommandResult execute(Model model) {
    requireNonNull(model);
    return new CommandResult(CommandRecorder.getLastCommand());
}
```

Reply:
That was exactly my old implementation. I changed it because I need this method to throw this exception in order to let UI know that **HistoryCommand** is being executing, and thus change the textbox in GUI section. Otherwise it would not work because Command should not have access to UI.

Reply:
The other commands are using `return new CommandResult(string)` to display success text to the GUI. You might want to look into how CommandResult is being utilized.